pkg/deprecatetemplates/allowlist.go

```diff
@@ -0,0 +1,150 @@
+package deprecatetemplates
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+
+	jc "github.com/openshift/ci-tools/pkg/jobconfig"
+)
+
+const (
+	presubmit  = "presubmit"
+	postsubmit = "postsubmit"
+	periodic   = "periodic"
+	release    = "release"
+	unknown    = "unknown"
+)
+
+func isGenerated(job config.JobBase) bool {
+	_, generated := job.Labels[jc.ProwJobLabelGenerated]
+	return generated
+}
+
+func getKind(job config.JobBase) string {
+	if strings.HasPrefix(job.Name, "pull-ci") {
+		return presubmit
+	} else if strings.HasPrefix(job.Name, "branch-ci-") {
+		return postsubmit
+	} else if strings.HasPrefix(job.Name, "release-") {
+		return release
+	} else if strings.HasPrefix(job.Name, "periodic-") {
+		return periodic
+	}
+
+	// this is fine, it is best effort
+	// the allowlist does not need to be 100% precise category-wise
+	return unknown
+}
+
+type blockedJob struct {
+	Generated bool   `json:"generated"`
+	Kind      string `json:"kind"`
+}
+
+type blockedJobs map[string]blockedJob
+
+func (b blockedJobs) Has(job config.JobBase) bool {
```

Comment:
I don't see us ever caring about the values in this map, can this be a simple `string.Set` or did I miss it?

Reply:
They are not programatically used right now, correct. I had two reasons to have these:

1. I want to be able to add a check like "no generated jobs should be in the allowlist for a template X"
2. I want to be able to eyeball the allowlist and understand what kinds of jobs are involved (that's why they are serialized)

Reply:
So remove them now and re-add them once you need them? Right now its premature optimization (and because the methods are exported, the deadcode checker won't complain)

Reply:
I'd rather not. It's not premature optimization b/c of reason 2. It's also not some hypothetical futureproofing, I *know* I will want to use the fields as soon as next week when we migrate all generated jobs for a template and I will want to add a "no generated jobs for this template" check.